### PR TITLE
Drop some deprecated APIs

### PR DESCRIPTION
**Pull request by kgaillot**
This breaks API backward compatibility and will not be backported to 2.1

### CODE REVIEW

**Comment by kgaillot** [comment]
@nrwahl2 and so it begins. low priority

**Comment by nrwahl2** [comment]
ahhhhhhhh :)

File: include/crm/pengine/pe_types_compat.h [hunk 1]
```diff
@@ -230,6 +230,7 @@ typedef struct pe_action_wrapper_s pe_action_wrapper_t;
 //! \deprecated Use pcmk_node_t instead
 typedef struct pe_node_s node_t;
 
+// @COMPAT This is used by sbd (as of at least 1.5.2)
```
Comment (nrwahl2) [inline include/crm/pengine/pe_types_compat.h:203]: I don't see this or `pe_resource_t` in sbd v1.5.2 code

**Review by nrwahl2 — COMMENTED** [review]
Woohoo!

File: include/crm/pengine/pe_types_compat.h (hunk 1 above)
Comment (kgaillot) [reply include/crm/pengine/pe_types_compat.h:203]: Ah these comments were left over from earlier work. Nope they are not used

Comment (kgaillot) [reply include/crm/pengine/pe_types_compat.h]: Dropped comments and added commits to delete those two types

**Review by nrwahl2 — APPROVED** [review]

**Comment by kgaillot** [comment]
retest this please

**Comment by nrwahl2** [comment]
```
14:54:57  make[5]: Entering directory '/srv/workspace/maker_pacemaker-pipeline_PR-3470/pacemaker/po'
14:54:57  make[5]: *** No rule to make target '../lib/pengine/common.c', needed by 'pacemaker.pot-update'.  Stop.
```

**Comment by kgaillot** [comment]
> ```
> 14:54:57  make[5]: Entering directory '/srv/workspace/maker_pacemaker-pipeline_PR-3470/pacemaker/po'
> 14:54:57  make[5]: *** No rule to make target '../lib/pengine/common.c', needed by 'pacemaker.pot-update'.  Stop.
> ```
> 
> Remove from `po/POTFILES.in`

Ah, didn't even think of that. I wonder why it didn't show up locally. Added commit on top.

**Review by nrwahl2 — APPROVED** [review]

**Pull request merged**